config/clusters/2i2c/dask-staging.values.yaml

```diff
@@ -37,13 +37,20 @@ basehub:
         JupyterHub:
           authenticator_class: cilogon
         CILogonOAuthenticator:
+          scope:
+            - "email"
+            - "profile"
           oauth_callback_url: "https://dask-staging.2i2c.cloud/hub/oauth_callback"
           # Only show the option to login with Google
           shown_idps:
-            - https://accounts.google.com/o/oauth2/auth
+            - http://accounts.google.com/o/oauth2/auth
+            - http://github.com/login/oauth/authorize
```

Comment:
> When you declare scope, which is a List, I expect the default to be replaced - not appended or similar. Due to that, I think the default values provided here: https://github.com/jupyterhub/oauthenticator/pull/575/files#diff-6f11d268e268a0d36dc3e95a40ff2b7b2785a842fa1a561df1e79be53172106cR94, seems like they will be overridden.

This is true. There are however, scopes that are appended too, in [the validation step of this config](
https://github.com/jupyterhub/oauthenticator/blob/6bf2d22b119ea60a96579be4fb86690288ceaebc/oauthenticator/cilogon.py#L103-L116). This is because `openid` is [a CILogon requirement](https://www.cilogon.org/oidc#h.p_PEQXL8QUjsQm) and the `org.cilogon.userinfo` is needed if we want to use information about the identity provider that was used to login, which we do when we specify an `allowed_idps` configuration.
So, in this case, the "real" scope list, because we also have `allowed_idps` specified, is actually `[email, profile, openid, org.cilogon.userinfo]`.

> I'm also uncertain about email and/or profile in scope. You have provided profile only for some CILogon -> GitHub setups. Do you have an overview of when and why you need certain scopes requested for misc authenticators behind CILogon?

This list [here](https://www.cilogon.org/oidc#h.p_PEQXL8QUjsQm) from CILogon presents the claims that **might** be present when requesting a certain scope. These are [all standard oidc claims](https://openid.net/specs/openid-connect-core-1_0.html#StandardClaims)

The thing is that CILogon only guarantees that the `sub` and `iss` claim will **always** be present in the response received after authenticating the user. The others are subject to the information received from the identity providers (Google, GitHub, institutions providers etc) when requesting a certain scope.

The GitHub "nickname" is **only** present as a claim (called `preffered_username`) in the auth response from GitHub if the `profile` scope is requested. Otherwise, we cannot grab that and use it as the hub username.
**I'm not sure of this**, but I believe, this claim used to be provided as part of one of the default scopes (maybe `org.cilogon.userinfo` as listed as "possible" [here](https://cilogon.org/.well-known/openid-configuration)) and GitHub stopped providing it sometime recent, as I remember being able to use `preffered_username` without explicitly requesting the `profile` scope.